app/scripts/modules/kubernetes/src/v2/manifest/selector/ManifestSelector.tsx

```diff
@@ -248,15 +272,25 @@ export class ManifestSelector extends React.Component<IManifestSelectorProps, IM
     this.setStateAndUpdateStage({ selector: this.state.selector });
   };
 
+  public handleKindsChange = (kinds: string[]): void => {
+    this.state.selector.kinds = kinds;
```

Comment:
This does not mutate the state object since its editing existing properties which is good. but I think it might be more idiomatic and easy to read if the edits were made to a copy of the state object and passed along to `setState()` rather than editing it directly.

Reply:
Yeah, the way this component handles state is a bit wonky and probably deserves a thorough refactor. Going to leave as-is for now, only because this is the pattern across all of the pre-existing methods (see `handleClusterChange`, `handleCriteriaChange`, `handleNamespaceChange`, etc. above) but I will try to revisit this next time we have to edit this component.